changegen/generator.py

```diff
@@ -504,24 +538,95 @@ def generate_changes(
 
         new_nodes = []
         new_ways = []
+        new_relations = []
 
-        if isinstance(wgs84_geom, sg.MultiLineString):
-            raise NotImplementedError("MultiLineString not implemeted.")
+        if isinstance(wgs84_geom, sg.MultiLineString) or isinstance(
+            wgs84_geom, sg.MultiPolygon
+        ):
+            raise NotImplementedError("Multi geometries not supported.")
         if isinstance(wgs84_geom, sg.LineString):
             ways, nodes = _generate_ways_and_nodes(
-                wgs84_geom, ids, feat_tags, intersection_db
+                wgs84_geom,
+                ids,
+                feat_tags,
+                intersection_db,
+                max_nodes_per_way=max_nodes_per_way,
             )
             new_nodes.extend(nodes)
             new_ways.extend(ways)
             _global_node_id_all_ways.extend(chain.from_iterable([w.nds for w in ways]))
+        if isinstance(wgs84_geom, sg.Polygon):
+            # simple polygons can be treated like Ways.
+            if len(wgs84_geom.interiors) == 0:
```

Comment:
A relation also needs to be created If the exterior is more than 2000 nodes, is this handled somehow?

Reply:
I didn't think of that. Does that mean anytime I split a long way into multiple ways I should create a relation joining them? I can see the case of a polygon exterior wanting to be a part of a multipolygon(?), that makes sense -- but what about regular Ways that are longer than 2000?

Reply:
Anytime a polygon gets split you need to create a multipolygon relation, which is confusing, because it's not a multipolygon, it's just a single polygon. But there is no polygon relation. Then you just apply the same tags to the relation that would have been applied to the way, and have no tags on the way.

It's more complicated for non-polygon ways. If you split a trail, the most correct thing to do is probably to create a route relation, but i think most of our current code knows how to deal with rejoining things if the rest of the attributes match, so the route relation is not strictly needed, and would probably be a waste of time to implement now.